packages/lib/src/components/internal/Address/components/AddressSearch.tsx

```diff
@@ -15,15 +16,33 @@ export type OnAddressLookupType = (
     }
 ) => Promise<void>;
 
+export type OnAddressSelectedType = (
+    value: string,
+    actions: {
+        resolve: (value: AddressLookupItem) => void;
+        reject: (reason?: any) => void;
+    }
+) => Promise<void>;
+
 interface AddressSearchProps {
     onAddressLookup?: OnAddressLookupType;
-    onSelect: any; //TODO
+    onAddressSelected?: OnAddressSelectedType;
+    onSelect: (addressItem: AddressData) => void;
     onManualAddress: any;
     externalErrorMessage: string;
     hideManualButton: boolean;
+    addressSearchDebounceMs?: number;
 }
 
-export default function AddressSearch({ onAddressLookup, onSelect, onManualAddress, externalErrorMessage, hideManualButton }: AddressSearchProps) {
+export default function AddressSearch({
```

Comment:
This component could be unit tested - you could test the callbacks and passing back values to them.
It might be worth to do it, as it has potential to be used in multiple places.

Reply:
Done!